gordon_janitor_gcp/http_client.py

```diff
@@ -0,0 +1,261 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Module to interact with Google APIs via asynchronous HTTP calls.
+
+Only service account (JSON Web Tokens/JWT) authentication is currently
+supported. To setup a service account, follow `Google's docs <https://
+cloud.google.com/iam/docs/creating-managing-service-account-keys>`_.
+
+To use:
+
+.. code-block:: python
+
+    keyfile = '/path/to/service_account_keyfile.json'
+    client = AIOGoogleHTTPClient(keyfile)
+    resp = await client.request('get', 'http://api.example.com/foo')
+
+"""
+
+import asyncio
+import datetime
+import http.client
+import json
+import logging
+import urllib.parse
+
+import aiohttp
+from google.oauth2 import _client
+from google.oauth2 import service_account
+
+from gordon_janitor_gcp import exceptions
+
+
+DEFAULT_REQUEST_HEADERS = {
+    'X-Goog-API-Client': 'custom-aiohttp-gcloud-python/3.6.2 gccl',
+    'Accept-Encoding': 'gzip',
+    'User-Agent': 'custom-aiohttp-gcloud-python',
+    'Authorization': '',
+}
+MAX_REFRESH_ATTEMPTS = 2
+REFRESH_STATUS_CODES = (http.client.UNAUTHORIZED,)
```

Comment:
Mimicking what Google does in their [auth lib](https://github.com/spotify/gordon-janitor-gcp/pull/1/).